Implement 'teardown'

w-miller opened the pull request:
* Refactor Ansible structure to allow commands
* Implement a new 'teardown' command in addition to 'deploy'
* Add documentation about override files for configuration

Commit:
Add documentation about override file

Conflicts:
	README.md
	ansible/host_vars/localhost

Commit:
Fix Markdown title formatting

Commit:
Restructure playbooks for multiple commands

Much of the logic in the existing 'deploy' playbooks could be reused for
a teardown command, so it didn't make sense to duplicate this. Instead,
rename/restructure playbooks such that they refer to an area of
responsibility, rather than an action.

Also, make some changes to allow a new 'teardown' action.

Commit:
Add 'teardown' action to veth-pair role

Commit:
Add 'teardown' action to virtualbmc-domain role

Commit:
Fix veth_pair_state typo

Commit:
Ensure physical_network.yml handles teardown

Commit:
Fix veth-pair role for teardown

* Don't delete the OVS bridge specified, as it will already be deleted
  when the role is called.
* Clarify the behaviour of the role for each value of `cmd` in the
  README.
* Succintify main.yml.

Commit:
Rename virtualbmc-domain internal variable

The variable 'cmd' conflicts with the Tenks global variable.

Commit:
Allow teardown of Nova flavors

Commit:
Allow teardown in Ironic enrolment

Commit:
Fix multiple 'when' clauses

markgoddard commented on ansible/override.yml.example:
```diff
@@ -0,0 +1,23 @@
+# This is an example override file, which can be used as a template for your
+# own override file.
+
+node_types:
```
Each of these could do with a comment describing what it is. Could just copy from host_vars.

markgoddard reviewed (commented):
Generally looks sensible. Typically when tearing things down we need to 'unwind' the creation process, to ensure dependent things are cleaned up in the correct order. Is that required here?

For example, is there going to be a problem if we remove veth pairs before the VM into which they are plugged?

How about if there were two top level playbooks, deploy.yml and cleanup.yml/teardown.yml. Teardown could import the playbooks in the reverse order. They could also use set_fact to set cmd automatically in each case.

WDYT? Perhaps it's not required.

Commit:
Split deployment and teardown playbooks

Create deploy.yml and teardown.yml which mostly run the same playbooks in
opposite orders. Also document the new structure.

Commit:
Improve comments for override file variables

Commit:
Merge branch 'master' into 2teardown

w-miller commented:
The reverse order is a good plan. While I didn't encounter any problems with running the teardown in the same order, it at least conceptually makes more sense to do it in reverse order, as you say, and will hopefully prevent future problems.

Updated and ready for re-review.

Commit:
Add names to set_fact tasks

Commit:
Add updated diagram
